mantidproject/mantid
# ISIS SANS - Standardise on front rear

**Description of work.**
The ISIS SANS Group has agreed to standardise on Front / Rear to replace HAB / LAB respectively. This is because only one instrument uses the latter, whilst others use the former. This PR:

- Renames HAB / LAB to Front / Rear in the UI
- Renamed HAB / LAB in the GUI models / presenters. But does not go into the algorithm layer / enums due to size.
- Fixes a bug introduced in #27895 which prevents us seeing the beam centre values updating (the method went AWOL from the presenter), as it's needed in this PR to validate everything still works

**Report to:** @smk78 

**To test:**
- Download the ISIS Training Data, find the `loqdemo` folder
- Open ISIS SANS
- Set the mask file to the `.txt` or `.toml` file in loqdemo
- Set the batch file to the .csv in the same folder
- Select the first row, hit load
- Switch the beam centre finder on the left
- The rows should be called Front / Rear in the UI. In one drop down there will be `main-detector` and `HAB`, this is expected as it comes from the IDF for LOQ
- Check the Front / Rear values exist, hit run. After it completes the values in Rear should have updated to the new centre in the text log
- Change from `main-detector` to `HAB` hit run again. Check the values in front have updated this time.

Part of #30514 

The fix to the beam centre value was introduced this release, so doesn't require release notes



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
